core/src/main/java/org/n52/iceland/binding/kvp/KvpBinding.java

```diff
@@ -180,7 +189,12 @@ public String getRequestParameterValue(Map<String, String> map) throws OwsExcept
         Decoder<AbstractServiceRequest<?>, Map<String, String>> decoder = getDecoder(k);
         LOGGER.trace("Using {} to decode paramers: {}", decoder, Arrays.toString(parameterValueMap.entrySet().toArray()));
         if (decoder != null) {
-            return decoder.decode(parameterValueMap);
+            AbstractServiceRequest<?> request = decoder.decode(parameterValueMap);
+            if (includeOriginal) {
+                request.setOriginalRequest(Joiner.on(Constants.QUERSTIONMARK_STRING)
```

Comment:
Will do - but if the ``QUESTIONMARK_STRING`` is removed... then many constants in https://github.com/52North/iceland/blob/develop/core/src/main/java/org/n52/iceland/util/Constants.java should be up for discussion.